nifi-commons/nifi-utils/src/test/java/org/apache/nifi/processor/TestFormatUtils.java

```diff
@@ -103,4 +111,41 @@ public void testFormatDataSize() {
 
         assertEquals(String.format("181%s9 TB", decimalFormatSymbols.getDecimalSeparator()), FormatUtils.formatDataSize(200_000_000_000_000d));
     }
+
+    @Test
+    public void testParseInstant() throws Exception {
+        TimeZone current = TimeZone.getDefault();
+        TimeZone.setDefault(TimeZone.getTimeZone("Europe/Kiev"));
+        try {
+            checkSameResultsWithSimpleDateFormat("yyyy-MM-dd HH:mm:ss", "2020-01-01 02:00:00", null, "2020-01-01T00:00:00");
+            checkSameResultsWithSimpleDateFormat("yyyy-MM-dd HH:mm:ss", "2020-01-01 01:00:00", "Europe/Warsaw", "2020-01-01T00:00:00");
+            checkSameResultsWithSimpleDateFormat("yyyy-MM-dd HH:mm:ss Z", "2020-01-01 01:00:00 +0100", null, "2020-01-01T00:00:00");
+            checkSameResultsWithSimpleDateFormat("yyyy-MM-dd", "2020-01-01", null, "2019-12-31T22:00:00");
+            checkSameResultsWithSimpleDateFormat("HH:mm:ss", "03:00:00", null, "1970-01-01T00:00:00");
+            checkSameResultsWithSimpleDateFormat("yyyy-MMM-dd", "2020-may-01", null, "2020-04-30T21:00:00");
+        } finally {
+            TimeZone.setDefault(current);
+        }
+    }
+
+    private void checkSameResultsWithSimpleDateFormat(String pattern, String parsedDateTime, String zone, String expectedUtcDateTime) throws Exception {
+        LocalDateTime expectedDateTime = LocalDateTime.parse(expectedUtcDateTime);
+
+        // reference implementation
+        SimpleDateFormat sdf = new SimpleDateFormat(pattern, Locale.US);
+        if (zone != null) {
+            sdf.setTimeZone(TimeZone.getTimeZone(zone));
+        }
+        Instant simpleDateFormatResult = sdf.parse(parsedDateTime).toInstant();
+        assertEquals(expectedDateTime, simpleDateFormatResult.atZone(ZoneOffset.UTC).toLocalDateTime());
```

Comment:
You've right but I don't want to hardcode instant as a millis from 1970 - it would obfuscate test cases. I've changed the logic that now result isn't in UTC but expected value is converted to instant using UTC. It will be more readable?